src/ore/src/future.rs

```diff
@@ -38,6 +38,15 @@ use tokio::time::{self, Duration, Instant};
 use crate::panic::CATCHING_UNWIND_ASYNC;
 use crate::task::{self, JoinHandleExt};
 
+/// Whether or not to run the future in `run_in_task_if` in a task.
```

Comment:
would be helpful to add some context to why this struct/choice is necessary in the code. If I were to come across this in the future without the context of this PR I think it'd be fairly confusing to understand it's purpose. Perhaps something like:

```
/// Whether or not to run the future in `run_in_task_if` in a task.
/// Useful in contexts where futures may be starved and cause inadvertent
/// failures in I/O-sensitive operations, such as when called within timely
/// operators.
```